src/main/java/WeatherstationEndpoint.java

```diff
@@ -1,52 +1,71 @@
 import javax.ws.rs.*;
 import javax.ws.rs.core.MediaType;
+import java.rmi.ServerException;
 
 /**
  * Provides an endpoint for weather data from multiple weather-stations located at various locations.
  * Possible input formats and outputs formats are described in the corresponding methods documentation.
  */
-@Path("weatherstation")
+@Path("{weatherstation}")
+
 public class WeatherstationEndpoint {
     private WeatherstationDAO database = new MongoDBDAO();
 
     /**
-     * Inserts a text/JSON formatted String provided in the 'message' parameter to the database.
-     * @param message contains the application/json formatted String to be inserted in the database.
-     * @return returns 'Database updated' if the procedure succeeded.
+     * Inserts a text/JSON formatted String provided in the 'message' parameter to the database in the collection / table with the name provided in the 'weatherstationName' parameter.
+     * @param message contains the text/JSON formatted String to be inserted in the database.
+     * @param weatherstationName the name of the weatherstation that provided the data.
+     * @return returns 'Database updated' if the procedure succeeded or the error message.
      */
     @POST
     @Consumes(MediaType.APPLICATION_JSON)
-    public String setWeatherstationData(String message)
+    public String setWeatherstationData(String message, @PathParam("weatherstation") String weatherstationName)
     {
+        System.out.println("Kald setWeatherstationData(message , weatherstationName) med " + message + " og " + weatherstationName);
+        try
+        {
+            return database.setWeatherstationData(message , weatherstationName);
+        }
+        catch(ServerException se)
+        {
+            return se.getMessage();
```

Comment:
Do NOT convert an exception to a string - this returns 200 OK - which is wrong

Reply:
ExceptionMapper is the rigth solution. Should return a meaningful errorCodehttps://howtodoinjava.com/jersey/jaxrs-jersey-exceptionmapper/